firedrake/assemble.py

```diff
@@ -416,7 +416,9 @@ def visitor(e, *operands):
 
     if tensor:
         update_tensor(result, tensor)
-    return result
+        return tensor
+    else:
+        return result
```

Comment:
I think this should work:

```suggestion
        return tensor
    return result
```

Reply:
Isn't this completely equivalent?

Reply:
Oh I misread that. I thought `update_tensor` had been removed